lib/gsockaddr.c

```diff
@@ -84,7 +84,7 @@ g_sockaddr_new(struct sockaddr *sa, int salen)
 }
 
 GSockAddr *
-g_sockaddr_new_from_peer_fd(gint fd)
+g_sockaddr_new_from_peer_name(gint fd)
```

Comment:
Unfortunately this change breaks installer generation. PE modules expects `g_sockaddr_new_from_peer_fd` to be available. I could fixup that in PE modules, but I am just not sure if the new name is better. Actually, the GSockAddr is still created from fd, and not from string. In that sense, the original name was better.

Reply:
The socket address is traditionally called "name" in the various socket APIs. See getsockname() & getpeername(). Also, the next patch in the series introduces a g_sockaddr_new_from_local_name(), which will use the same "fd".

Also, I understand the PE concerns, all I can offer is to  provide a patch against that code if you (as in One Identity) choose to publish them :)

Reply:
(just a note)
... where name means address and peer is an fd...
so we could just call it as `g_sockaddr_new_from_peer` (or `get_peer_gsockaddr`) and it's an internal detail that we get the peer address with `getpeername`.

So I think if we rename this function (and also rename it in the internal `ALTP` module), then I'd choose another name :)

Reply:
Does `name` refers to the output parameter `addr`, instead of the input parameter `fd`, both in case of `getsockname` and `getpeername`? Me feels more like getting the name, instead of getting something from the name. `

Anyhow, for now, I just created a pe patch with the new name. Also I needed to rebase the branch to master, because one of the pe unit tests fail. In case you need to change the code, could you rebase against master. It would help me with the followup a little.

Other than these, the installer is generated, so no other blocking problem so far. I try to find help and time to do the performance test next week.

Reply:
We are constructing a GSockAddr instance based on a socket peer's name, where socket is denoted by fd. Maybe `g_sockaddr_new_from_connection_peer()` would work better? Then the other variant would be `g_sockaddr_new_from_connection_local()`, which feels a bit strange and pretty long too.

Another solution would be not using a constructor at all, rather add a function `g_socket_get_local_name()` and `g_socket_get_peer_name()` to gsocket.c

You know what, I just implemented the last paragraph and it does feel much better!

What do you think?

Reply:
Looks good to me. Thank you!